src/h3/mod.rs

```diff
@@ -456,6 +464,9 @@ pub enum Event {
 
     /// Data was received.
     Data(Vec<u8>),
+
+    // Stream was closed,
+    StreamClosed(),
```

Comment:
```suggestion
    Finished,
```

So a couple of things: 1. I don't think you need `()` since you are not actually wrapping another type (this also means that matching this event would be somewhat more readable) 2. `Finished` seems like a better name (and shorter), but YMMV.